Assets/Scripts/SaveSystem/SaveableGameObject.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using UnityEngine.SceneManagement;
+
+namespace SaveSystem
+{
+    //Component that is added to GameObjects that have other components (that implements ISaveable) that need to be saved
+    [ExecuteInEditMode]
+    public class SaveableGameObject : MonoBehaviour
```

Comment:
Wait so you need to add this component to the gameobject to be saved?

Reply:
yes

Reply:
that searches the game object for all saveable components/variables